code/game/objects/effects/spawners/random/bedsheet.dm

```diff
@@ -0,0 +1,64 @@
+/obj/effect/spawner/random/bedsheet
+	name = "random dorm bedsheet"
+	icon_state = "random_bedsheet"
+	loot = list(/obj/item/bedsheet,
+		/obj/item/bedsheet/blue = 8,
```

Comment:
also fix the list formatting on this (is it intentional for the base bedsheet type to not have a weight associated with it? please add one otherwise)